cmd/dockerfile/digests.go

```diff
@@ -0,0 +1,110 @@
+package dockerfile
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/buildsafedev/bsf/cmd/styles"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/spf13/cobra"
+)
+
+var DGCmd = &cobra.Command{
+	Use:     "digests",
+	Short:   "Replace Dockerfile image tags with immutable digests",
+	Aliases: []string{"dg"},
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) < 1 {
+			fmt.Println(styles.HintStyle.Render("hint:", "run `bsf dockerfile digests <Dockerfile>` to replace image tags with digests"))
+			os.Exit(1)
+		}
+
+		dockerfile := args[0]
+		file, err := os.ReadFile(dockerfile)
+		if err != nil {
+			fmt.Printf("Error opening Dockerfile: %v\n", err)
+			os.Exit(1)
+		}
+
+		var dgMap = map[string]string{}
+
+		dgMap, err = readByte(file)
+		if err != nil {
+			fmt.Printf("Error in readByte %v\n", err)
+			os.Exit(1)
+		}
+
+		dgMap, err = getDigest(dgMap)
+		if err != nil {
+			fmt.Printf("Error retrieving digest %v\n", err)
+			os.Exit(1)
+		}
+
+		updatedData, err := updateDockerfileWithDigests(file, dgMap)
+		if err != nil {
+			fmt.Printf("Error updating Dockerfile with digests: %v\n", err)
+			os.Exit(1)
+		}
+
+		if err := os.WriteFile(dockerfile, updatedData, 0644); err != nil {
+			fmt.Printf("Error writing updated Dockerfile: %v\n", err)
+			os.Exit(1)
+		}
+		fmt.Println("Changes made in")
+		for _, data := range updatedData {
+			fmt.Println(string(data))
+		}
+	},
+}
+
+func getDigest(dgMap map[string]string) (map[string]string, error) {
+
+	for img := range dgMap {
+		dg, err := crane.Digest(img)
```

Comment:
Can we parallelise this with a sync/wait group?

Reply:
Yup thought of this, to use goroutine let me try on this